src/ComplexType.php

```diff
@@ -102,27 +139,52 @@ protected function generateClass()
             $name = Validator::validateAttribute($member->getName());
             $typeHint = Validator::validateTypeHint($type);
 
-            $comment = new PhpDocComment();
-            $comment->setVar(PhpDocElementFactory::getVar($type, $name, ''));
-            $var = new PhpVariable('protected', $name, 'null', $comment);
-            $this->class->addVariable($var);
+            $property = new PropertyGenerator();
+            $property
+                ->setDefaultValue(new PropertyValueGenerator(null, ValueGenerator::TYPE_NULL))
+                ->setDocBlock(
+                    (new DocBlockGenerator())
+                        ->setTag(new VarTag($name, $type))
+                )
+                ->setFlags(PropertyGenerator::FLAG_PROTECTED)
+                ->setName($name);
+
+            $this->class->addPropertyFromGenerator($property);
 
             if (!$member->getNullable()) {
                 if ($type == '\DateTime') {
                     if ($this->config->get('constructorParamsDefaultToNull')) {
-                        $constructorSource .= '  $this->' . $name . ' = $' . $name . ' ? $' . $name . '->format(\DateTime::ATOM) : null;' . PHP_EOL;
+                        $constructorSource .= '$this->' . $name . ' = $' . $name . ' ? $' . $name . '->format(\DateTime::ATOM) : null;' . PHP_EOL;
                     } else {
-                        $constructorSource .= '  $this->' . $name . ' = $' . $name . '->format(\DateTime::ATOM);' . PHP_EOL;
+                        $constructorSource .= '$this->' . $name . ' = $' . $name . '->format(\DateTime::ATOM);' . PHP_EOL;
                     }
                 } else {
-                    $constructorSource .= '  $this->' . $name . ' = $' . $name . ';' . PHP_EOL;
+                    $constructorSource .= '$this->' . $name . ' = $' . $name . ';' . PHP_EOL;
+                }
+
+                $constructorParameter = new ParameterGenerator();
+                $constructorParameter->setName($name);
+                if (!empty($typeHint)) {
+                    $constructorParameter->setType($typeHint);
+                }
+                if ($type == '\DateTime' || $this->config->get('constructorParamsDefaultToNull')) {
+                    $constructorParameter->setDefaultValue(
+                        new ValueGenerator(null, ValueGenerator::TYPE_NULL)
+                    );
```

Comment:
This seems to be duplicated around [L108](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/pull/255/files#diff-07244fe9da4f9a298082444a2875a671R108). Could this be refactored?